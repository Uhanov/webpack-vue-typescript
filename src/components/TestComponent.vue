<template>
  <section>
    <span>{{count}}</span>
    <button v-on:click="addToCount">Recount it</button>
  </section>
</template>

<script lang="ts">
  import Vue from 'vue';
  export default Vue.extend({
    name: 'test-component',
    data () {
      return {
        count: 0
      }
    },
    props: {
      countStep: {
        type: Number,
        default: 0
      }
    },
    methods: {
      addToCount () {
        this.count+= this.countStep || 1
        this.$emit('count', this.count)
      }
    }
  })
</script>
