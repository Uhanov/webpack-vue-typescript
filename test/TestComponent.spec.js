import { shallowMount } from '@vue/test-utils'
import TestComponent from '@/components/TestComponent.vue'

describe('TestComponent', () => {
  const wrapper = shallowMount(TestComponent)
  const button = wrapper.find('button')

  it('should render correct contents', () => {
    expect(button.textContent).toMatchSnapshot();
  });

  it('counts right number', () => {
    wrapper.setProps({countStep: 20})
    button.trigger('click')
    expect(wrapper.vm.count).toBe(20)
  })
})
