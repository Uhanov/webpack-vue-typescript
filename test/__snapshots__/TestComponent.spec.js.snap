// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent should render correct contents 1`] = `undefined`;
